deployments/helm-chart/templates/controller-serviceaccount.yaml

```diff
@@ -10,4 +10,8 @@ metadata:
 imagePullSecrets:
 {{ toYaml .Values.controller.serviceAccount.imagePullSecrets }}
 {{- end }}
+{{- if .Values.controller.serviceAccount.imagePullSecretName }}
```

Comment:
If someone were to specify both `controller.serviceAccount.imagePullSecrets` and `controller.serviceAccount.imagePullSecrets` you'd end up with two `imagePullSecrets` blocks in the file. Should the behaviour be one takes precedence over the other? Maybe `controller.serviceAccount.imagePullSecretName` could have precedence since `controller.serviceAccount.imagePullSecrets` is being deprecated?

Reply:
Yeah it's good to protect against this.